src/base/bittorrent/torrenthandle.cpp

```diff
@@ -207,15 +206,28 @@ TorrentHandle::TorrentHandle(Session *session, const libtorrent::torrent_handle
     // download sequentially or have first/last piece priority enabled when
     // its resume data was saved. These two settings are restored later. But
     // if we set them to false now, both will erroneously not be restored.
-    if (!data.resumed || data.sequential)
-        setSequentialDownload(data.sequential);
-    if (!data.resumed || data.firstLastPiecePriority)
-        setFirstLastPiecePriority(data.firstLastPiecePriority);
+    if (!params.restored || params.sequential)
+        setSequentialDownload(params.sequential);
+    if (!params.restored || params.firstLastPiecePriority)
+        setFirstLastPiecePriority(params.firstLastPiecePriority);
 
-    if (!data.resumed && hasMetadata()) {
+    if (!params.restored && hasMetadata()) {
         if (filesCount() == 1)
             m_hasRootFolder = false;
     }
+
+    // "started" means "all initialization has completed and torrent has started regular processing".
+    // When torrent added/restored in "paused" state it become "started" immediately after construction.
+    // When it is added/restored in "resumed" state, it become "started" after it is really resumed
+    // (i.e. after receiving "torrent resumed" alert).
+    m_started = (params.restored && hasMetadata() ? isPaused() : params.paused);
+
+    if (!m_started) {
+        if (!params.restored || !hasMetadata())
```

Comment:
since comments are added, better add brackets to inner if() too.

Reply:
Ok, fixed.